lib/cors.js

```diff
@@ -7,21 +8,30 @@ const defs = {
   origin: '*'
 }
 
-const basics = opts => ({
+const basics = (opts, req) => ({
   'access-control-allow-credentials': opts.credentials || defs.credentials,
-  'access-control-allow-origin': opts.origin || defs.origin
+  'access-control-allow-origin': chooseOrigin(opts.origin || defs.origin, pathOr('', ['headers', 'origin'], req))
 })
 
-const cors = (app, opts={}) => req =>
+const cors = curry((app, opts={}) => req =>
```

Comment:
@pklingem, this won't do what you think it does.  Because the second arg has a default value, the `.length` of the function will only be `1`, which is the same as not currying it at all.

Reply:
ok, was gonna ask you if this presented a problem... will revert.